compiler/optimizer/VPHandlers.cpp

```diff
@@ -9829,42 +9807,39 @@ static TR::Node *constrainIfcmpeqne(OMR::ValuePropagation *vp, TR::Node *node, b
                                                    objectClass, cpIndexOrVftSlot, symRef->getOwningMethod(vp->comp()),
                                                    vp->comp(), false, useGetResolvedInterfaceMethod);
 
-        TR_ScratchList<TR_PersistentClassInfo> subClasses(vp->comp()->trMemory());
-
-        //in AOT or when CHOpts are disabled findClassInfoAfterLocking returns NULL unconditionally
-        //we should treat this case conservatively meaning we assume that there are more than two concrete classes in the given hierarchy
-        //most likely rvm will also be NULL in this case, but it is better to not rely on this implicit connection between different
-        //CH queries.
-        TR_PersistentClassInfo *objectClassInfo = vp->comp()->getPersistentInfo()->getPersistentCHTable()->findClassInfoAfterLocking(objectClass, vp->comp());
+                TR_ScratchList<TR_PersistentClassInfo> subClasses(vp->comp()->trMemory());
 
-        bool searchSucceeded = false;
+                //in AOT or when CHOpts are disabled findClassInfoAfterLocking returns NULL unconditionally
+                //we should treat this case conservatively meaning we assume that there are more than two concrete classes in the given hierarchy
+                //most likely rvm will also be NULL in this case, but it is better to not rely on this implicit connection between different
+                //CH queries.
+                TR_PersistentClassInfo *objectClassInfo = vp->comp()->getPersistentInfo()->getPersistentCHTable()->findClassInfoAfterLocking(objectClass, vp->comp());
 
-        if (objectClassInfo)
-          {
-          TR_ClassQueries::collectAllSubClasses(objectClassInfo, &subClasses, vp->comp());
-          subClasses.add(objectClassInfo);
-          searchSucceeded = true;
-          }
-
-      /*
-        the limited version of a transformation is enabled
-        Namely, we only do a transformation if there is no more than one instantiable class in the hierarchy
-        starting from objectClass. If there is more than one concrete class in hierarchy
-        overriding methods in objectClass and if any call was devirtualized by invariantargumentpreexisence downstream
-        we will produce the functionally incorrect code by swapping a profiled test with a method test as
-        the method test is insufficient for the devirtualizations done by invariantargumentpreexisence
-      */
-
-      if (searchSucceeded && numConcreteClasses(&subClasses) < 2 && rvm)
+                if (rvm && objectClassInfo)
```

Comment:
If `rvm` is null you can avoid calling `vp->comp()->getPersistentInfo()->getPersistentCHTable()->findClassInfoAfterLocking(objectClass, vp->comp());`

Reply:
Made change to this comment, also moved the declaration of `TR_ScratchList<TR_PersistentClassInfo> subClasses` inside `if (objectClassInfo)` block.